src/logics/ArrayTheory.cc

```diff
@@ -7,7 +7,9 @@
 #include "ArrayTheory.h"
 
 #include "ArrayHelpers.h"
-#include "DistinctRewriter.h"
+#include "Rewritings.h"
+
+using namespace opensmt::rewritings;
```

Comment:
My understanding is that it is a bad idea in header files, in .cpp files, it should not do any harm.
This is related to a general question, if we are gonna have sub-namespaces, or everything will be in a single `opensmt` namespace.

Reply:
Update: in header files it is a really bad idea, in cpp files IMO a bit bad. In general, it can still cause name clashes, even though only locally and not globally. It can become a problem if the source file becomes large. But for small files it is probably perfectly fine.

I vote for using sub-namespaces. However, in this case, I am not sure, because the fully scoped names fall back to `opensmt::rewritings::rewrite_*` which is IMO already a bit too verbose.